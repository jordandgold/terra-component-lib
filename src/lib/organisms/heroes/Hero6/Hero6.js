import React from "react";
import "./Hero6.scss";
import PropTypes from "prop-types";
import CTALinkSection from "../../../CTALinkSection/CTALinkSection";

const Hero6 = props => {
  const { title, text, ctas, image, imageSide } = props;
  const imageStyle = {
    background: `url(${image.url})`
  };
  if (imageSide === "right") {
    return (
      <section className="ter-hero-six ter-hero-six--right-image">
        <div className="ter-hero-six__content-container ter-hero-six__content-container--left">
          <h2 className="ter-hero-six__title">{title}</h2>
          <p className="ter-hero-six__text">{text}</p>
          <CTALinkSection ctas={ctas} />
        </div>
        <div
          style={imageStyle}
          className="ter-hero-six__image-container ter-hero-six__image-container--right-image"
        />
      </section>
    );
  } else {
    return (
      <section className="ter-hero-six">
        <div className="ter-hero-six__image-container" />
        <div className="ter-hero-six__content-container">
          <h2 className="ter-hero-six__title">{title}</h2>
          <p className="ter-hero-six__text">{text}</p>
          <CTALinkSection ctas={ctas} />
        </div>
      </section>
    );
  }
};

export default Hero6;

Hero6.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  text: PropTypes.string.isRequired,
  ctas: PropTypes.shape({
    ctaOne: PropTypes.shape({
      className: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
      link: PropTypes.string.isRequired
    }),
    ctaTwo: PropTypes.shape({
      className: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
      link: PropTypes.string.isRequired
    }),
    subCTA: PropTypes.shape({
      text: PropTypes.string.isRequired,
      link: PropTypes.string.isRequired
    })
  })
};
