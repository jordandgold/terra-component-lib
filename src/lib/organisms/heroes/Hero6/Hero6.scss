@import "../../../global-styles/terra.scss";

.ter-hero-6 {
  background-color: color("white");
  position: relative;
  &--right {
    .ter-hero-6__hero-image {
      right: 0;
      left: unset;
    }
    .ter-hero-6__content {
      grid-column-end: auto;
    }
  }
  &__hero-image {
    display: block;
    @media screen and (min-width: $breakpoint-sm) {
      background-size: cover;
      position: absolute;
      left: 0;
      top: 0;
      height: 100%;
      width: 54%;
    }
    img {
      width: 100%;
      height: auto;
      display: block;
    }
    &--3x2 {
      display: none;
      @media screen and (min-width: $breakpoint-md) {
        display: block;
      }
    }
    &--4x3 {
      display: block;
      @media screen and (min-width: $breakpoint-sm) {
        display: none;
      }
    }
    &--1x1 {
      display: none;
      @media screen and (min-width: $breakpoint-sm) and (max-width: $breakpoint-md) {
        display: block;
      }
    }
  }
  &__container {
    @include grid-container();
  }
  &__content {
    padding: spacing("md") 0;
    @include grid-column(12);
    @media screen and (min-width: $breakpoint-sm) {
      @include grid-column(5);
      grid-column-end: -1;
      padding: spacing("mega") 0;
    }
  }
  .ter-cta-section {
    align-self: flex-start;
  }
  .ter-header {
    font-size: font-size("xl");
  }
  .ter-sub-header {
    font-size: font-size("md");
    font-weight: font-weight("light");
    margin-bottom: spacing("md");
  }
  .ter-cta-section {
    @media (min-width: $breakpoint-xs) {
      max-width: 200px;
    }
    &__link {
      display: block;
      text-align: center;
      font-size: font-size("base");
      margin-top: spacing("xxs");
    }
  }
}
