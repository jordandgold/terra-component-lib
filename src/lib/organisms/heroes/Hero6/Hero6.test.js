import React from "react";
import { shallow } from "enzyme";
import Hero6 from "./Hero6";

describe("Hero6", () => {
  let wrapper;
  const heroSixContent = {
    images: {
      imageThreeByTwo: {
        url: "http://fpoimg.com/2400x1600?text=3:2",
        altText: "placeholder image"
      },
      imageFourByThree: {
        url: "http://fpoimg.com/1600x1200?text=4:3",
        altText: "placeholder image"
      },
      imageOneByOne: {
        url: "http://fpoimg.com/1600x1600?text=1:1",
        altText: "placeholder image"
      }
    },
    title: "Good design is innovative.",
    text:
      "Good design is aesthetic. Good design makes a product understandable. Good design is unobtrusive.",
    ctas: {
      ctaOne: {
        className: "ter-button--primary--1",
        link: "#",
        text: "Button"
      },
      ctaTwo: {
        className: "ter-button--secondary--1",
        link: "#",
        text: "Button"
      }
    }
  };

  beforeEach(() => {
    wrapper = shallow(
      <Hero6
        imageSide="left"
        images={heroSixContent.images}
        title="Good design is innovative."
        text={heroSixContent.text}
        ctas={heroSixContent.ctas}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
