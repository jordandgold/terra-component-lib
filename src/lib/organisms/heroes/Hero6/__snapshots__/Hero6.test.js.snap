// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero6 should match the snapshot 1`] = `
<section
  class="ter-hero-6 ter-hero-6--left"
>
  <figure
    class="ter-hero-6__hero-image ter-hero-6__hero-image--3x2"
    style={
      Object {
        "background": "url(http://fpoimg.com/2400x1600?text=3:2)",
      }
    }
  />
  <figure
    class="ter-hero-6__hero-image ter-hero-6__hero-image--1x1"
    style={
      Object {
        "background": "url(http://fpoimg.com/1600x1600?text=1:1)",
      }
    }
  />
  <figure
    class="ter-hero-6__hero-image ter-hero-6__hero-image--4x3"
  >
    <img
      src="http://fpoimg.com/1600x1200?text=4:3"
    />
  </figure>
  <div
    class="ter-hero-6__container"
  >
    <aside
      class="ter-hero-6__content"
    >
      <h2
        className="ter-hero-6__title"
      >
        Good design is innovative.
      </h2>
      <p
        className="ter-hero-6__text"
      >
        Good design is aesthetic. Good design makes a product understandable. Good design is unobtrusive.
      </p>
      <CTALinkSection
        ctas={
          Object {
            "ctaOne": Object {
              "className": "ter-button--primary--1",
              "link": "#",
              "text": "Button",
            },
            "ctaTwo": Object {
              "className": "ter-button--secondary--1",
              "link": "#",
              "text": "Button",
            },
          }
        }
      />
    </aside>
  </div>
</section>
`;
